src/Features/Core/Portable/CodeRefactorings/SyncNamespace/ISyncNamespaceService.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.SyncNamespace
+{
+    internal interface ISyncNamespaceService : ILanguageService
```

Comment:
why do we need a service...  is this called from multiple locations?

Reply:
Yes, currently there's language specific logic when fixing references to the types declared in changed namespace.

Reply:
Why do we need language specific logic?

Reply:
Right now it's because there's a bunch of syntax manipulation. Some of those potentially can be  made language agnostic. However, please note that I 'm planning to re-purpose this service to handle #30896, and VB will be supported as well. A language service is required because VB has very different namespace rules in certain cases. (for example, root namespace, modules, etc).